ui/perfherder/CompareSubtestDistributionView.jsx

```diff
@@ -0,0 +1,304 @@
+      return;
+    }
+    const numRuns = perfDatumList[subtestSignature].length;
+    const replicatePromises = perfDatumList[subtestSignature].map(value =>
+      PerfSeriesModel.getReplicateData({ job_id: value.job_id }),
+    );
+
+    return Promise.all(replicatePromises).then(
+      localReplicateData => {
+        let replicateValues = localReplicateData.concat.apply(
+          [],
+          localReplicateData.map(data => {
+            const testSuite = data.suites.find(
+              suite => suite.name === this.state.testSuite,
+            );
+            const subtest = testSuite.subtests.find(
+              subtest => subtest.name === this.state.subtest,
+            );
+            return subtest.replicates;
+          }),
+        );
+        // metrics-graphics doesn't accept "0" as x_accesor
+        replicateValues = replicateValues.map((value, index) => ({
+          replicate: (index + 1).toString(),
+          value,
+        }));
+
+        return {
+          replicateData,
+          drawingData: { numRuns, replicateValues },
+        };
+      },
+      () => {
+        replicateData.replicateDataError = true;
+      },
+    );
+  };
+
+  render() {
+    const {
+      originalProject,
+      newProject,
+      originalResultSet,
+      newResultSet,
+      originalReplicateError,
+      newReplicateError,
+      dataLoading,
+      platform,
+      testName,
+      graphOneSpecs,
+      graphTwoSpecs,
+      baseDrawingData,
+      newDrawingData,
+    } = this.state;
+    const { originalRevision, newRevision } = this.props.$stateParams;
+    const baseData =
+      baseDrawingData && baseDrawingData.replicateValues
+        ? baseDrawingData.replicateValues
+        : [];
+    const newData =
+      newDrawingData && newDrawingData.replicateValues
+        ? newDrawingData.replicateValues
+        : [];
+    // debugger; // eslint-disable-line no-debugger
+    return (
+      originalRevision &&
+      newRevision && (
+        <div className="container-fluid">
+          {dataLoading ? (
+            <div>
+              Loading all results, please wait a minute...
+              <img alt="dancing cat" src={dancing_cat} />
+            </div>
+          ) : (
+            <div>
+              <h2>
+                {platform}: {testName} replicate distribution
+              </h2>
+              <RevisionInformation
+                originalProject={originalProject}
+                originalRevision={originalRevision}
+                originalResultSet={originalResultSet}
+                newProject={newProject}
+                newRevision={newRevision}
+                newResultSet={newResultSet}
+              />
+              &nbsp;&nbsp;
+              {!(originalReplicateError && newReplicateError) ? (
+                <div>
+                  {graphOneSpecs && (
+                    <Graph specs={graphOneSpecs} data={baseData} />
+                  )}
+                  {graphTwoSpecs && (
```

Comment:
I think you need to verify whether there is any data or errors (should add some sort of error handling if an endpoint fails) before you render this Graph.

Reply:
I think it'd simplify the code if you created one generic class component as a wrapper for the graphs so each graph fetches its own data in that a component (but the query parameters will be validated in this parent component).